src/field_poison.c

```diff
@@ -90,9 +90,9 @@ static void Task_TryFieldPoisonWhiteOut(u8 taskId)
         {
             // Battle facilities have their own white out script to handle the challenge loss
 #ifdef BUGFIX
-            if (InBattlePyramid() || InBattlePike() || InTrainerHillChallenge())
+            if (CurrentBattlePyramidLocation() != PYRAMID_LOCATION_NONE || InBattlePike() || InTrainerHillChallenge())
 #else
-            if (InBattlePyramid() | InBattlePike() || InTrainerHillChallenge())
+            if (CurrentBattlePyramidLocation() | InBattlePike() || InTrainerHillChallenge())
```

Comment:
These should be the same, otherwise it looks like the difference is significant for the bug fix.